src/main/java/seedu/ui/AddUI.java

```diff
@@ -0,0 +1,176 @@
+        int lectureSerial = SERIAL_STARTING;
+        int tutorialSerial = SERIAL_STARTING;
+        int labSerial = SERIAL_STARTING;
+        int index = Math.max(lecture.size(), Math.max(tutorial.size(), laboratory.size()));
+
+        for (int i = 0; index > i; i++) {
+            if (lecture.size() > i) {
+                Lesson lesson = lecture.get(i);
+                String detail = printLessonInfo(lectureSerial, lesson);
+                lectureLessons.add(detail);
+                boolean isGap = classNumberGap(lecture, lesson);
+                if (isGap) {
+                    lectureLessons.add("_______________________________________   |   ");
+                    lectureSerial++;
+                }
+            }
+            if (tutorial.size() > i) {
+                Lesson lesson = tutorial.get(i);
+                String detail = printLessonInfo(tutorialSerial, lesson);
+                tutorialLessons.add(detail);
+                boolean isGap = classNumberGap(tutorial, lesson);
+                if (isGap) {
+                    tutorialLessons.add("_______________________________________   |   ");
+                    tutorialSerial++;
+                }
+            }
+            if (laboratory.size() > i) {
+                Lesson lesson = laboratory.get(i);
+                String detail = printLessonInfo(labSerial, lesson);
+                laboratoryLessons.add(detail);
+                boolean isGap = classNumberGap(laboratory, lesson);
+                if (isGap) {
+                    laboratoryLessons.add("_______________________________________");
+                    labSerial++;
+                }
+            }
+        }
+        TextUi.printLessonHeader(lecture, tutorial, laboratory);
+        index = Math.max(lectureLessons.size(), Math.max(tutorialLessons.size(), laboratoryLessons.size()));
+        for (int j = 0; index > j; j++) {
+            String output = "";
+            if (lectureLessons.size() > j && isExist(lecture)) {
+                output = output.concat(lectureLessons.get(j));
+            } else if (tutorialLessons.size() > j && isExist(lecture)) {
+                for (int i = 0; GAP > i; i++) {
+                    output = output.concat(SPACE);
+                }
+            }
+            if (tutorialLessons.size() > j) {
+                output = output.concat(tutorialLessons.get(j));
+            } else if (laboratoryLessons.size() > j && isExist(tutorial)) {
+                for (int i = 0; GAP > i; i++) {
+                    output = output.concat(SPACE);
+                }
+            }
+            if (laboratoryLessons.size() > j) {
+                output = output.concat(laboratoryLessons.get(j));
+            }
+            System.out.println(output);
+        }
+
+        if (isExist(lecture)) {
+            String select = TextUi.getLessonCommand(LECTURE);
+            int indexOfLesson = Integer.parseInt(select) - BALANCE_ARRAY;
```

Comment:
What happens if parseInt() throws a NumberFormatException here? I could be wrong but it does not seem like anything prevents the user from inputting a non integer in getLessonCommand nor is there any catch statement for NumberFormatException.